### ISSUE TITLE
Fullpageos Proxy settings

### ISSUE THREAD

**Issue by lukeyboytas1** [guysoft/FullPageOS#316]
HI - Any help on how to add proxy server settings?

**Comment by guysoft** [comment]
If you press ctrl+t on a keyboard/vnc you should get a nornal chromium window where you can access the browser settings and set up a proxy

**Comment by lukeyboytas1** [comment]
That is an awesome start.... Now I get this ( see pic)
![20200216_200437](https://user-images.githubusercontent.com/30971091/74601952-bdd1aa00-50f7-11ea-83ef-0d8361cd7abe.jpg)

**Comment by guysoft** [comment]
Oh, seems like when you paste that message in to google you get a commandline argument to do this ([source](https://www.internalpointers.com/post/run-chrome-chromium-proxy-settings-linux)).
So it seems you will need to add this argument in the file [``/home/pi/scripts/start_chromium_browser``](https://github.com/guysoft/FullPageOS/blob/devel/src/modules/fullpageos/filesystem/home/pi/scripts/start_chromium_browser)
The command arguments are set is this file and you can change them, including adding that proxy server flag.

**Comment by stappersg** [comment]
@lukeyboytas1 how about any feedback on the additional advice from @guysoft ?

**Comment by guysoft** [comment]
Ill close this issue if there is no reply

**Comment by lukeyboytas1** [comment]
Sorry guys - this is over my head - I dont understand adding arguments?

**Comment by guysoft** [comment]
@lukeyboytas1 There is a file with a line that writes the command used to start chromium. That command has list of arguments. [Here is an article on this in the chromium project](https://www.chromium.org/developers/how-tos/run-chromium-with-flags).
If you edit that file you can change that line to have different settings. As explained in the chromium tab you took a photo on.
 You need to have basic understanding how to use a raspberrypi to edit a file there. Once you do it should use the proxy from that line that you edit.
The path of the file in the filesystem is ``/home/pi/scripts/start_chromium_browser`` as I provided above.

**Comment by stappersg** [comment]
> Oh, seems like when you paste that message in to google you get a commandline argument to do this ([source](https://www.internalpointers.com/post/run-chrome-chromium-proxy-settings-linux)).
> So it seems you will need to add this argument in the file [`/home/pi/scripts/start_chromium_browser`](https://github.com/guysoft/FullPageOS/blob/devel/src/modules/fullpageos/filesystem/home/pi/scripts/start_chromium_browser)
> The command arguments are set is this file and you can change them, including adding that proxy server flag.

Yes, adding ` --proxy-server="protocol://hostname:port" ` to the many options on `chromium` in `/home/pi/scripts/start_chromium_browser` works for me.

**Comment by stappersg** [comment]
@lukeyboytas1 thanks for raising the question.  I'm glad found this issue.  **and** the advice from guysoft.

For is it fine that this issue gets closed.  (@lukeyboytas1 you have to power to do so)

**Comment by 60999** [comment]
--proxy-server="socks5://127.0.0.1:1080" 
 --proxy-server="http://127.0.0.1:1081"

**Comment by 60999** [comment]
in
`--proxy-server="protocol://hostname:port"`
<protocol>could be one of them：

"http", "socks", "socks4", "socks5".

**Comment by stritti** [comment]
Proxy could also be defined global in Rasbian, which also helps to run apt updates:

```sh
cd /etc/apt/apt.conf.d
sudo touch 10proxy
sudo nano 10proxy
```

```sh
//file=/etc/apt/apt.conf.d/10proxy
Acquire::http::Proxy "http://127.0.0.1:8080";
Acquire::https::Proxy "http://127.0.0.1:8080";
```
